docs/tutorials/writing_custom_configaudit_policies.md

```diff
@@ -0,0 +1,218 @@
+define custom policies and associate them with applicable Kubernetes resources to extend basic configuration audit
+functionality.
+
+This tutorial will walk through the process of creating and testing a new configuration audit policy that fails whenever
+a Kubernetes resource doesn't specify `app.kubernetes.io/name` or `app.kubernetes.io/version` labels.
+
+## Writing a Policy
+
+To define such policy, you must first define its metadata. This includes setting a unique identifier, title, severity
+(`CRITICAL`, `HIGH`, `MEDIUM`, `LOW`), descriptive text, and remediation steps. In Rego it's defined as the
+`__rego_metadata__` rule, which defines the following composite value:
+
+```opa
+package starboard.policy.k8s.custom
+
+__rego_metadata__ := {
+	"id": "recommended_labels",
+	"title": "Recommended labels",
+	"severity": "LOW",
+	"type": "Kubernetes Security Check",
+	"description": "A common set of labels allows tools to work interoperably, describing objects in a common manner that all tools can understand.",
+	"recommended_actions": "Take full advantage of using recommended labels and apply them on every resource object.",
+	"url": "https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/",
+}
+```
+
+Note that the `recommended_labels` policy in scoped to the `starboard.policy.k8s.custom` package to avoid naming
+collision with built-in policies that are pre-installed with Starboard.
+
+Once we've got our metadata defined, we need to create the logic of the policy, which is done in the `deny` or `warn`
+rule.
+
+```opa
+recommended_labels := [
+	"app.kubernetes.io/name",
+	"app.kubernetes.io/version",
+]
+
+deny[res] {
+	provided := {label | input.metadata.labels[label]}
+	required := {label | label := recommended_labels[_]}
+	missing := required - provided
+	count(missing) > 0
+	msg := sprintf("You must provide labels: %v", [missing])
+	res := {"msg": msg}
+}
+```
+
+These matches are essentially Rego assertions, so anyone familiar with writing rules for OPA or other tools that use
+Rego should find the process familiar. In this case, it’s pretty straightforward. We subtract the set of labels
+specified by the `input` resource object from the set of recommended labels. The resulting set is stored in the variable
+called `missing`. Finally, we check if the `missing` set is empty. If not, the `deny` rule fails with the appropriate
+message.
+
+The `input` document is set by Starboard to a Kubernetes resource when the policy is evaluated. For Pods, it would look
+something like the following listing:
+
+```json
+{
+  "apiVersion": "v1",
+  "kind": "Pod",
+  "metadata": {
+    "name": "nginx",
+    "labels": {
+      "run": "nginx"
+    }
+  },
+  "spec": {
+    "containers": [
+      {
+        "name": "nginx",
+        "image": "nginx:1.16",
+      }
+    ]
+  }
+}
+```
+
+Therefore, provided labels can be retrieved with the following Rego expression:
```

Comment:
I would make it more explicit -- "The labels set on the pod resource above can be retrieved with the following Rego expression:"

Reply:
Done